### PR TITLE
Extend library path to have multiarch directory

### PR DESCRIPTION
**Pull request by jacobperron**
This fixes issues finding libraries installed by external packages using GNUInstallDirs.
For example, CycloneDDS, see: https://github.com/ros2/rmw_cyclonedds/issues/182

This is a revival of a patch introduced in https://github.com/ros2/ros_workspace/pull/2 and subsequently removed in https://github.com/ros2/ros_workspace/pull/10.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,7 +43,18 @@ if(WIN32)
...
+  execute_process(COMMAND gcc -dumpmachine OUTPUT_VARIABLE MULTIARCH_TRIPLET OUTPUT_STRIP_TRAILING_WHITESPACE)
+  # register multiarch information for .dsv generation
+  if(APPLE)
+    set(LIBRARY_PATH_ENV_VAR "DYLD_LIBRARY_PATH")
+  else()
+    set(LIBRARY_PATH_ENV_VAR "LD_LIBRARY_PATH")
+  endif()
+  set(
+    AMENT_CMAKE_ENVIRONMENT_HOOKS_DESC_multiarch_library_paths
+    "prepend-non-duplicate;${LIBRARY_PATH_ENV_VAR};${MULTIARCH_TRIPLET}")
```
Comment (jacobperron) [inline CMakeLists.txt]: @dirk-thomas Is this the right approach (generating a .dsv file)? Or should we instead add a .dsv file explicitly to this repository?

Comment (dirk-thomas) [reply CMakeLists.txt]: The last part needs to contain the relative path from the install prefix. I assume it lacks `lib/` at the moment?

Comment (jacobperron) [reply CMakeLists.txt]: Yes, you are correct.  Fixed in 81289b6

**Review by dirk-thomas — APPROVED** [review]

**Comment by jacobperron** [comment]
@nuclearsandwich I'll let you confirm that this fixes the issue before merging.

**Review by nuclearsandwich — APPROVED** [review]
Looks good and works when building a deb locally.

Thanks!

**Comment by nuclearsandwich** [comment]
@seanyen I'm going to merge this as is, but I suspect this environment hook isn't necessary for your Windows packaging builds as the package which install into a multiarch LIB directory probably don't do so when building on Windows.

**Pull request merged**